libbeat/metric/system/process/process.go

```diff
@@ -462,138 +121,223 @@ func (procStats *Stats) Init() error {
 		cgReader, err := cgroup.NewReaderOptions(procStats.CgroupOpts)
 		if err == cgroup.ErrCgroupsMissing {
 			logp.Warn("cgroup data collection will be disabled: %v", err)
-			procStats.EnableCgroups = false
 		} else if err != nil {
 			return errors.Wrap(err, "error initializing cgroup reader")
 		}
 		procStats.cgroups = cgReader
 	}
-
 	return nil
 }
 
-// Get fetches process data which matches the provided regexes from the host.
-func (procStats *Stats) Get() ([]common.MapStr, error) {
+// Get fetches the configured processes and returns a formatted map, plus the root event
+func (procStats *Stats) Get() ([]mb.Event, error) {
+	//If the user hasn't configured any kind of process glob, return
 	if len(procStats.Procs) == 0 {
 		return nil, nil
 	}
 
-	pids, err := Pids()
+	// actually fetch the PIDs from the OS-specific code
+	pidMap, plist, err := procStats.FetchPids()
+
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch the list of PIDs")
+		return nil, errors.Wrap(err, "error gathering PIDs")
 	}
+	// We use this to track processes over time.
+	procStats.ProcsMap = pidMap
 
-	var processes []Process
-	newProcs := make(ProcsMap, len(pids))
+	// filter the process list that will be passed down to users
+	procStats.includeTopProcesses(plist)
 
-	for _, pid := range pids {
-		process := procStats.getSingleProcess(pid, newProcs)
-		if process == nil {
-			continue
+	// This is a holdover until we migrate this library to metricbeat/internal
+	// At which point we'll use the memory code there.
+	var totalPhyMem uint64
+	if procStats.host != nil {
+		memStats, err := procStats.host.Memory()
+		if err != nil {
+			procStats.logger.Warnf("Getting memory details: %v", err)
+		} else {
+			totalPhyMem = memStats.Total
 		}
-		processes = append(processes, *process)
+
 	}
-	procStats.ProcsMap = newProcs
 
-	processes = procStats.includeTopProcesses(processes)
-	procStats.logger.Debugf("Filtered top processes down to %d processes", len(processes))
+	//Format the list to the MapStr type used by the outputs
+	procs := make([]mb.Event, 0, len(plist))
```

Comment:
Why is this `mb.Event`? Shouldn't this be `beat.Event`?
Please do not import anything from Metricbeat or other Beats into Libbeat.